docs/source/gateway.rst

```diff
@@ -0,0 +1,745 @@
+    * **name**: The name of the model to use. For example, ``gpt-3.5-turbo`` for `OpenAI`'s ``GPT-3.5-Turbo`` model.
+    * **config**: Contains any additional configuration details required for the model. This includes specifying the API base URL and the API key.
+
+Here's an example of a route configuration:
+
+.. code-block:: yaml
+
+    routes:
+        - name: completions
+          type: chat/completions
+          model:
+            provider: openai
+            name: gpt-3.5-turbo
+            config:
+              openai_api_base: https://api.openai.com/v1
+              openai_api_key: $OPENAI_API_KEY
+
+In the example above, a request sent to the completions route would be forwarded to the
+``gpt-3.5-turbo`` model provided by ``openai``.
+
+The routes in the configuration file can be updated at any time, and the `MLflow Gateway` will
+automatically update its available routes without requiring a restart. This feature provides you
+with the flexibility to add, remove, or modify routes as your needs change. It enables 'hot-swapping'
+of routes, providing a seamless experience for any applications or services that interact with the `MLflow Gateway`.
+
+When defining routes in the configuration file, ensure that each name is unique to prevent conflicts.
+Duplicate route names will raise an Exception.
+
+Providers
+---------
+The `MLflow Gateway` is designed to support a variety of model providers.
+A provider represents the source of the machine learning models, such as `OpenAI`, `Anthropic`, and so on.
+Each provider has its specific characteristics and configurations that are encapsulated within the model part of a route in the `MLflow Gateway`.
+
+Within each model block in the configuration file, the provider field is used to specify the name
+of the provider for that model. This is a string value that needs to correspond to a provider the `MLflow Gateway` supports.
+
+Here's an example of a provider configuration within a route:
+
+.. code-block:: yaml
+
+    routes:
+        - name: chat
+          type: llm/v1/chat
+          model:
+            provider: openai
+            name: gpt-4
+            config:
+              openai_api_base: https://api.openai.com/v1
+              openai_api_key: $OPENAI_API_KEY
+
+In the above configuration, ``openai`` is the `provider` for the model.
+
+As of now, the `MLflow Gateway` supports the following providers:
+
+* **openai**: This is used for models offered by `OpenAI <https://platform.openai.com/>`_.
+* **anthropic**: This is used for models offered by `Anthropic <https://docs.anthropic.com/claude/docs>`_.
+
+More providers are being added continually. Check the latest version of the `MLflow Gateway` for the
+most up-to-date list of supported providers.
+
+Remember, the provider you specify must be one that the `MLflow Gateway` supports. If the provider
+is not supported, the Gateway will return an error when trying to route requests to that provider.
+
+Model Names
```

Comment:
I think this section should be called `Models` and should be moved above the section called `Providers`. I think `Providers` should be a subsection of `Models`.